src/main/java/QueryServlet.java

```diff
@@ -218,12 +244,128 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
+          .put("S0201_125E,S0201_126E,S0201_119E", ImmutableList.of(add, percent))
+          .put("K200701_005E,K200701_006E", ImmutableList.of(add))
+          .put("K200701_004E,K200701_005E,K200701_006E", ImmutableList.of(add, add))
+          .put("S0201_124E,S0201_125E,S0201_126E,S0201_119E", ImmutableList.of(add, add, percent))
+          .put(
+              "S0701_C03_012E,S0701_C04_012E,S0701_C05_012E,S0701_C01_012E",
+              ImmutableList.of(add, add, percent))
+          .put(
+              "S0701_C03_013E,S0701_C04_013E,S0701_C05_013E,S0701_C01_013E",
+              ImmutableList.of(add, add, percent))
+          .put(
+              "P014001,P014021,P014022,P014042,P014043",
+              ImmutableList.of(rightSubtract, rightSubtract, rightSubtract, rightSubtract))
+          .build();
+
+  private String reformatDataArray(String data, String dataIdentifier, boolean isCountyQuery)
+      throws InvalidObjectException {
+    if (!dataRowToReformatFunction.containsKey(dataIdentifier)) {
+      return data; // This data doesn't need reformatting
+    }
+    if ((dataRowToReformatFunction.get(dataIdentifier).size() + 1)
+        != dataIdentifier.split(",").length) {
+      // We should have enough functions listed to apply them to each group
+      // of two rows - (a, b) and then (their product, c), etc.
+      throw new InvalidObjectException(
+          "There aren't the right number of functions to apply to these data columns.");
+    }
+
+    // Convert from the census API's JSON string to a Java matrix
+    Gson gson = new Gson();
+    Type dataArrayType = new TypeToken<ArrayList<List<String>>>() {}.getType();
+    ArrayList<List<String>> originalDataArray = gson.fromJson(data, dataArrayType);
+    ArrayList<List<String>> newDataArray = new ArrayList<List<String>>();
+
+    // Add back the the first column, which is the string identifier of the location,
+    // and the second column, the first numerical value
+    for (int i = 0; i < originalDataArray.size(); i++) {
+      List<String> originalDataRow = originalDataArray.get(i);
+      List<String> newDataRow = new ArrayList<String>();
+      newDataRow.add(originalDataRow.get(0));
+      newDataRow.add(originalDataRow.get(1));
+      newDataArray.add(newDataRow);
+    }
+
+    // Combine the original data, skipping the header row, by iterating over the functions
+    // given and successively combining the next datapoint with the previous result
+    List<BiFunction> numberCombiners = dataRowToReformatFunction.get(dataIdentifier);
+    for (int i = 0; i < numberCombiners.size(); i++) {
+      BiFunction numberCombiner = numberCombiners.get(i);
+      for (int j = 1; j < originalDataArray.size(); j++) {
+        List<String> originalDataRow = originalDataArray.get(j);
+        List<String> newDataRow = newDataArray.get(j);
+        if (newDataRow.get(1) == null || newDataRow.get(1).startsWith("-")) {
```

Comment:
we use get(1) here because newDataRow only consists of two columns where the first column is the state name right? Like ["California", 120,000] and we would only want to access the 120,000